edx_event_bus_kafka/publishing/event_producer.py

```diff
@@ -131,35 +130,15 @@ def get_producer_for_signal(signal: OpenEdxPublicSignal, event_key_field: str) -
         remote-config (and in particular does not result in mixed cache/uncached configuration).
         This complexity is being deferred until this becomes a performance issue.
     """
-    if schema_registry_url := getattr(settings, 'SCHEMA_REGISTRY_URL', None):
-        schema_registry_config = {
-            'url': schema_registry_url,
-            'basic.auth.user.info': f"{getattr(settings, 'SCHEMA_REGISTRY_API_KEY', '')}"
-                                    f":{getattr(settings, 'SCHEMA_REGISTRY_API_SECRET', '')}",
-        }
-    else:
-        warnings.warn("Cannot configure event-bus-kafka: Missing setting SCHEMA_REGISTRY_URL")
+    schema_registry_client = create_schema_registry_client()
+    if schema_registry_client is None:
         return None
 
-    if bootstrap_servers := getattr(settings, 'KAFKA_BOOTSTRAP_SERVERS', None):
-        producer_settings = {
-            'bootstrap.servers': bootstrap_servers,
-        }
-    else:
-        warnings.warn("Cannot configure event-bus-kafka: Missing setting KAFKA_BOOTSTRAP_SERVERS")
+    producer_settings = load_common_settings()
+    if producer_settings is None:
         return None
 
-    if getattr(settings, 'KAFKA_API_KEY', None) and getattr(settings, 'KAFKA_API_SECRET', None):
-        producer_settings.update({
-            'sasl.mechanism': 'PLAIN',
-            'security.protocol': 'SASL_SSL',
-            'sasl.username': settings.KAFKA_API_KEY,
-            'sasl.password': settings.KAFKA_API_SECRET,
-        })
-
-    schema_registry_client = SchemaRegistryClient(schema_registry_config)
     signal_serializer = get_serializer(signal)
-
```

Comment:
I'm not sure why the linter is saying there's an issue on line 147, but you'll want to restore this empty line to make it happy. (Blank line before a `def`.)